server/lib/pbench/process_tb.py

```
@@ -0,0 +1,113 @@
+        and token options
+
+        Args -
+            controller -- the name of the controller to be associated with the
+                          tar ball
+            tb -- path of the tar ball
+            token -- generated authorised token for Pbench user
+
+        Importance of this function is while running tests we get the
+        ability to mock this function and test it easily
+        """
+
+        res = subprocess.run(
+            ["pbench-results-push", controller, tb, token], capture_output=True
+        )
+        error = res.stderr
+        if res.returncode > 0:
+            raise RuntimeError(error)
+
+    def process_tb(self) -> Results:
+        """Searches for tarballs in the configured receive directory and
+        uploads them to the Pbench server"""
+
+        # Check for results that are ready for processing: version 002 agents
+        # upload the MD5 file as xxx.md5.check and they rename it to xxx.md5
+        # after they are done with MD5 checking so that's what we look for.
+        self.logger.info("{}", self.config.TS)
+        token = self.config.get("pbench-server", "put-token")
+        nstatus = ""
+        ntotal = ntbs = 0
+
+        for tbmd5 in sorted(self.receive_dir.glob("**/*.tar.xz.md5")):
+            ntotal += 1
+
+            # Extracts full pathname of tarball from the name of hashfile
+            # by trimming .md5 suffix
+            tb = Path(str(tbmd5)[0 : -len(".md5")])
+            tbdir = tb.parent
+            controller = tbdir.name
+
+            try:
+                self.results_push(controller, tb, token)
+            except Exception as e:
+                self.logger.error(
+                    "{}: Unexpected Error while running Agent's 'pbench-result-push' command: {}",
+                    self.config.TS,
+                    e,
+                )
+                continue
+
+            try:
+                os.remove(tbmd5)
+                os.remove(tb)
```

Consider moving this to separate try blocks so the tar ball is not stranded.

---

Just to be clear:  Peter, you're hoping that the removal (attempt) of `tb` will be completed before reporting on the failure to remove `tbmd5`, right?  So, we need not only separate `try` blocks, but the second `try` block needs to be in a `finally` clause of the first....

---

Nevermind:  I see that we're just eating the exception after logging it, so we don't need the `finally` block.